src/sys/socket/mod.rs

```diff
@@ -829,6 +829,11 @@ pub enum ControlMessageOwned {
     #[cfg_attr(docsrs, doc(cfg(feature = "net")))]
     Ipv6RecvErr(libc::sock_extended_err, Option<sockaddr_in6>),
 
+    /// `SOL_TLS` messages of type `TLS_GET_RECORD_TYPE`, containing the TLS message content type,
+    /// normally one of change_cipher_spec(20), alert(21), handshake(22) (for TLS 1.3
+    /// resumption tickets), application_data(23)
+    TlsGetRecordType(u8),
```

Comment:
Should this be an enum type?

Reply:
I briefly considered it but, for it to be future-proof it'd need an "Unknown" variant. I don't mind it happening, but I feel like there'd be _many_ back-and-forths on the PR until everyone's happy with it.

Reply:
I don't think this is good enough.  Matching against magic numbers isn't Rusty at all.  I think you should add the various types, including an Unknown variant.

Reply:
As you wish — there's now an enum with the variants I know/care about.